crates/hir-expand/src/name.rs

```diff
@@ -112,6 +112,18 @@ impl Name {
         Name::new_inline("[missing name]")
     }
 
+    /// Generates a new name which is only equal to itself, by incrementing a counter. Due
+    /// its implementation, it should not be used in things that salsa considers, like
+    /// type names or field names, and it should be only used in names of local variables
+    /// and labels and similar things.
+    pub fn generate_new_name() -> Name {
+        use std::sync::atomic::{AtomicUsize, Ordering};
+        static CNT: AtomicUsize = AtomicUsize::new(0);
+        let c = CNT.fetch_add(1, Ordering::Relaxed);
+        // FIXME: Currently a `__RA_generated_name` in user code will break our analysis
+        Name::new_text(format!("__RA_geneated_name_{c}").into())
+    }
```

Comment:
Can you put a FIXME on this that we should really move to interned symbols (with support for unique name generation)? A similar comment already exists on `missing` but thats also missing a FIXME. The fact that we use `SmolStr` for names is something I really really dislike (which becomes apparent here)

Reply:
This function should be removed entirely when we resolve paths in hir lowering.